apps/src/craft/designer/craft.js

```diff
@@ -0,0 +1,913 @@
+    },
+    levelVerificationTimeout: levelConfig.levelVerificationTimeout,
+    isDaytime: levelConfig.isDaytime,
+    groundPlane: levelConfig.groundPlane,
+    usePlayer: !doNotShowPlayer,
+    useScore: levelConfig.useScore,
+    entities: levelConfig.entities,
+    isEventLevel: levelConfig.isEventLevel,
+    groundDecorationPlane: levelConfig.groundDecorationPlane,
+    actionPlane: levelConfig.actionPlane,
+    fluffPlane: fluffPlane,
+    playerStartPosition: levelConfig.playerStartPosition,
+    playerStartDirection: levelConfig.playerStartDirection,
+    playerName: Craft.getCurrentCharacter(),
+    assetPacks: levelAssetPacks,
+    gridDimensions: levelConfig.gridWidth && levelConfig.gridHeight ?
+        [levelConfig.gridWidth, levelConfig.gridHeight] :
+        null,
+    // eslint-disable-next-line no-eval
+    verificationFunction: eval('[' + levelConfig.verificationFunction + ']')[0], // TODO(bjordan): add to utils
+    // eslint-disable-next-line no-eval
+    failureCheckFunction: eval('[' + levelConfig.failureCheckFunction + ']')[0], // TODO(bjordan): add to utils
+    // eslint-disable-next-line no-eval
+    timeoutResult: eval('[' + (levelConfig.timeoutVerificationFunction || `function() { return false; }`) + ']')[0],
+  });
+};
+
+Craft.minAssetsForLevelWithCharacter = function (levelNumber) {
+  return Craft.minAssetsForLevelNumber(levelNumber)
+      .concat([Craft.characterAssetPackName(Craft.getCurrentCharacter())]);
+};
+
+Craft.minAssetsForLevelNumber = function (levelNumber) {
+  if (Craft.initialConfig.level.isEventLevel) {
+    return ['allAssetsMinusPlayer'];
+  }
+
+  switch (levelNumber) {
+    case 1:
+      return ['levelOneAssets'];
+    case 2:
+      return ['levelTwoAssets'];
+    case 3:
+      return ['levelThreeAssets'];
+    default:
+      return ['allAssetsMinusPlayer'];
+  }
+};
+
+Craft.afterLoadAssetsForLevel = function (levelNumber) {
+  if (Craft.initialConfig.level.isEventLevel) {
+    return ['allAssetsMinusPlayer'];
+  }
+
+  // After level loads & player starts playing, kick off further asset downloads
+  switch (levelNumber) {
+    case 1:
+      // can disable if performance issue on early level 1
+      return Craft.minAssetsForLevelNumber(2);
+    case 2:
+      return Craft.minAssetsForLevelNumber(3);
+    default:
+      // May want to push this to occur on level with video
+      return ['allAssetsMinusPlayer'];
+  }
+};
+
+Craft.earlyLoadAssetsForLevel = function (levelNumber) {
+  switch (levelNumber) {
+    case 1:
+      return Craft.minAssetsForLevelNumber(levelNumber);
+    default:
+      return Craft.minAssetsForLevelWithCharacter(levelNumber);
+  }
+};
+
+Craft.niceToHaveAssetsForLevel = function (levelNumber) {
+  switch (levelNumber) {
+    case 1:
+      return ['playerSteve', 'playerAlex'];
+    default:
+      return ['allAssetsMinusPlayer'];
+  }
+};
+
+/** Folds array B on top of array A */
+Craft.foldInArray = function (arrayA, arrayB) {
```

Comment:
Could this be a util function (in this file or elsewhere) instead of living on Craft?